src/logger/LoggerFactory.cpp

```diff
@@ -25,7 +25,7 @@ namespace sql
 {
 namespace mariadb
 {
-  Shared::Logger LoggerFactory::NO_LOGGER= (NO_LOGGER ? NO_LOGGER : Shared::Logger(new NoLogger()));
+  std::shared_ptr<Logger> LoggerFactory::NO_LOGGER = nullptr;
```

Comment:
Is there real need to change Shared::Logger to its real type? I don't think so. Beside there is probably no need to assign it to nullptr - I think default constructor will do the same thing, but for cheaper.